File: code/server/src/main/java/minihearthstone/entities/cards/effects/LifeStealEffect.java
package minihearthstone.entities.cards.effects;

import minihearthstone.entities.cards.Card;
import minihearthstone.entities.gameboard.GameBoard;
import minihearthstone.entities.heroes.Hero;

import javax.persistence.Entity;

@Entity
public class LifeStealEffect extends AbstractEffect {

    public LifeStealEffect() {
        setName("LifeStealEffect");
    }

    @Override
    public boolean applyOnChoiceEffect(Card card, Long playerId, GameBoard gameBoard) {
        return true;
    }

    @Override
    public boolean applyPreActionEffect(Card card, Long playerId, String targetCardUniqueId, GameBoard gameBoard) {
        return true;
    }

    @Override
    public boolean applyInActionPreCondEffect(Card card, Long playerId, GameBoard gameBoard) {
        log.info("LifeStealEffect -> Applying applyInActionPreCondEffect() | return true");
        return true;
    }

    @Override
    public boolean applyInActionPostCondEffect(Card card, Long playerId, GameBoard gameBoard) {
        log.info("LifeStealEffect -> Applying applyInActionPostCondEffect()");
        Hero hero = gameBoard.getOpponentHero(playerId);
        hero.setLifepoints(hero.getLifepoints() + card.getDamage());
        return true;
    }

}

File: code/client/src/app/card/effect/lifesteal-effect.ts
import { AbstractEffect } from './abstract-effect';

export class LifeStealEffect extends AbstractEffect {

    public LifeStealEffect() {
        this.name = 'LifeStealEffect';
    }

    public applyPreActionEffect(cardUniqueId: string, clientId: number, targetCardUniqueId: string, targetCardEffectName: string): boolean {
        console.log('LifeStealEffect -> Applying applyInActionPreCondEffect() | return true');
        return true;
    }
}
